common/lib/xmodule/xmodule/js/src/video/03_video_player.js

```diff
@@ -822,14 +815,18 @@ function (HTML5Video, Resizer) {
 
     function updatePlayTime(time, skip_seek) {
         var videoPlayer = this.videoPlayer,
-            duration = this.videoPlayer.duration(),
+            endTime = this.videoPlayer.duration(),
             youTubeId;
 
+        if (this.config.endTime) {
+            endTime = Math.min(this.config.endTime, endTime);
```

Comment:
Is it possible to create a function for getting the applicable `endTime`? This logic is repeated in three places at least.